bin/git-summary

```diff
@@ -216,9 +216,9 @@ print_summary_by_line() {
 print_summary() {
   if [ "$OUTPUT_STYLE" == "tabular" ]; then
     tabular_headers="# Repo $SP Age $SP Last active $SP Active on $SP Commits $SP Uncommitted $SP Branch"
-    echo -e "$tabular_headers\n$project $SP $(repository_age) $SP $(last_active) $SP $(active_days $commit) days $SP $(commit_count $commit) $SP $(uncommitted_changes_count) $SP $(current_branch_name)" | column -t -s "$COLUMN_CMD_DELIMTER"
+    echo -e "$tabular_headers\n$project $SP $(repository_age) $SP $(last_active) $SP $(active_days "$commit") days $SP $(commit_count "$commit") $SP $(uncommitted_changes_count) $SP $(current_branch_name)" | column -t -s "$COLUMN_CMD_DELIMTER"
```

Comment:
The quote is missing intently. We have `shellcheck disable=SC2086` mark in `active_days` and other similar functions.

Reply:
To me it doesn't look like the `shellcheck disable=SC2086` should even be in `active_days`. And actually, looking at all the ShellCheck disable directives, the ones that deal with `$commit` look incorrect. So to me it seems we are dealing with a larger issue that has been surpressed.

From the top, `git-summary` has a synposis of `git-summary [--line] [--dedup-by-email] [<committish>]`, and accepts a _single_ `committish` (even if it accepted multiple committishes, it wouldn't make sense anyways since the underlying git subcommands acccept only a single commitish). The `$commit` variable is either assigned `HEAD` or `$*` (which corresponds to `<committish>`. The `$*` looks fishy itself, but doesn't need to be changed.

Concerning the `active_days` function, the only argument passed to it is `$commit`, which is eventually passed to `git log`. In other functions, `$commit` is passed to `git shortlog`. All these git subcommands have the commonality of accepting a single `[<revision range>]`.

`[<revision range>]` is defined in `gitrevisions(7)`, and some valid revisions include things that contain spaces, like `HEAD^{/fix nasty bug}`, `HEAD@{5 minutes ago}`. So by not quoting `$commit`, it word splits on committishes that include spaces.

Reply:
However, there are cases in which commit may be empty and word splitting is intended - so those of course will have to be handled as well